exercises/zebra-puzzle/.meta/src/reference/java/ZebraPuzzle.java

```diff
@@ -0,0 +1,108 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+class ZebraPuzzle {
+    private String waterDrinker = "";
+    private String zebraOwner = "";
+
+    ZebraPuzzle() {
+        this.solve();
+    }
+
+    private static boolean isJustRightOf(int a, int b) {
+        return a - 1 == b;
+    }
+
+    private static boolean nextTo(int a, int b) {
+        return isJustRightOf(a, b) || isJustRightOf(b, a);
+    }
+
+    private void solve() {
+        int[] houses = new int[]{1,2,3,4,5};
+        int first = 1;
+        int middle = 3;
+
+        perms(houses).forEach(colors -> {
+            int red = colors[0], green = colors[1], ivory = colors[2], yellow = colors[3], blue = colors[4];
+
+            if (!isJustRightOf(green, ivory)) return;
```

Comment:
I prefer to always include curly brackets with `if` statements. Otherwise they're harder to spot and it's easier for people modifying the code later to accidentally introduce bugs (see our [policies doc](https://github.com/exercism/java/blob/master/POLICIES.md#adhere-to-best-practices)) 🙂